components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/js/design-view/design-grid.js

```diff
@@ -2292,7 +2292,11 @@ define(['require', 'log', 'jquery', 'backbone', 'lodash', 'designViewUtils', 'dr
             var self = this;
             var elementId;
             if (isCodeToDesignMode !== undefined && !isCodeToDesignMode) {
-                elementId = self.getNewAgentId();
+                if (stream !== undefined && stream.getId() !== undefined) {
```

Comment:
Do we need to do type check for undefined? Can't we do like, 

```js
elementId = stream && stream.getId() ? stream.getId() : self.getNewAgentId();
```

Reply:
+1